### ISSUE TITLE
Frontend: Error: Request timed out

### ISSUE THREAD

**Issue by ajnavarro** [gnolang/gnochess#86]
```
Uncaught (in promise) Error: Request timed out
    ye ws.js:119
    setTimeout handler*RA</Ae.prototype.sendRequest/WSProvider</</ie< ws.js:117
    WSProvider ws.js:116
    Be ws.js:33
    ye ws.js:14
    6 ws.js:8
    6 ws.js:4
    sendRequest ws.js:104
    GnoWSProvider ws.js:73
    Be ws.js:48
    ye ws.js:29
    1 ws.js:23
    1 ws.js:19
    evaluateExpression ws.js:69
    getGame actions.ts:276
    isGameOngoing actions.ts:265
    checkOngoingTimer gameboard.ts:228
    setInterval handler*intervalCheckForOngoingGame gameboard.ts:227
    startGame gameboard.ts:56
    value sevejs.umd.js:1
    _clickOnCtrl1 gameoptions.ts:318
    ye gsap-core.js:1970
    render gsap-core.js:2320
    render gsap-core.js:2257
    _lazySafeRender gsap-core.js:192
    updateRoot gsap-core.js:2695
    Ye gsap-core.js:1315
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
    Ye gsap-core.js:1310
10 ws.js:119:39

```

**Comment by alexiscolin** [comment]
@ajnavarro I think this one comes from Hugo server, because it looses the connection from local webserver to browser for a while. It should be only visible in localhost and not in prod.

**Comment by alexiscolin** [comment]
Nope, it comes from the tm2-js-client lib

**Comment by zivkovicmilos** [comment]
> This request timeout is from the WS provider in the client libs, it happens when the request is not served by the node in a specified time interval (by default 15s, this can be configured of course when you create the WS provider, both the TM2 one and the Gno one)

I've tried increasing the timeout, but that simply prolonged the problem.

I'm suspecting there is a hanging request somewhere, that triggers this timeout eventually (the node seems to be serving all game-related requests fine)

**Comment by jinoosss** [comment]
@zivkovicmilos 
It seems to me that if the request ID in WS is the same, the response is lost.

I think we need a more unique ID when creating a WS request.
Hard to guarantee that the time is unique when processing asynchronously.

https://github.com/gnolang/tm2-js-client/blob/main/src/provider/utility/requests.utility.ts#L15

**Comment by zivkovicmilos** [comment]
@jinoosss 

You are absolutely right.

I went back and looked over [this segment of the codebase](https://github.com/gnolang/tm2-js-client/blob/main/src/provider/websocket/ws.ts#L98). It explains why some requests are not being served, even when the node is serving them 🤦‍♂️ 

I am going to do a release with a switch to UUID, which should eliminate this issue. I'm keeping you posted on what we find 🙏

**Comment by zivkovicmilos** [comment]
Temporarily resolved by #110
